File: Licao06-Webservice/E01.02-DAO-WebserviceLocalSync/aplicativo/app/src/main/java/br/edu/ifsp/hto/htoipdm/applazycolumn/data/model/ApiResponse.kt
package br.edu.ifsp.hto.htoipdm.applazycolumn.data.model

import com.google.gson.annotations.SerializedName

data class ApiResponse<T>(
    @SerializedName("data_response")
    val dataResponse: T?,
    val message: String,
    @SerializedName("response_code")
    val responseCode: Long
)

File: Licao08-AplicativoCompleto/backend/app/model/api_response.py
class ApiResponse:
    def __init__(self, dataResponse, message: str | None, response_code: int):
        self.message = message
        self.dataResponse = dataResponse
        self.response_code = response_code

    def to_dict(self):
        return {
            "message": self.message,
            "data_response": self.dataResponse,
            "response_code": self.response_code,
        }
